File: java/1115.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.lang.NullPointerException;

class Main{
    private int ca, cb;

    Main(){};
    Main(int a, int b){
        this.ca = a;
        this.cb = b;
    }
    
    public int getCa(){
        return this.ca;
    }
    
    public int getCb(){
        return this.cb;
    }

    public void printQuadrante(){
        if(this.ca > 0 && this.cb > 0)
            System.out.println("primeiro");
        else if(this.ca > 0 && this.cb < 0)
            System.out.println("quarto");
        else if(this.ca < 0 && this.cb > 0)
            System.out.println("segundo");
        else if(this.ca  < 0 && cb < 0)
            System.out.println("terceiro");
    }

    public static String[] getData() throws NullPointerException{
        String[] data = null;
        try{
            BufferedReader buffer = new BufferedReader(new InputStreamReader(System.in));
            data = buffer.readLine().split(" ");
        } catch(IOException ioe){
            System.err.println("Error occurred: "+  ioe.getMessage());
        }
        return data;
    }

    public static void main(String[] args) throws IOException, NullPointerException{
        int a, b;
        do{
            String [] data = getData();
            a = Integer.parseInt(data[0]);
            b = Integer.parseInt(data[1]);
            Main m = new Main(a, b);
            m.printQuadrante();
            }while(a != 0 && b != 0);
    }
}

File: cplusplus/1115.cpp
#include<iostream>
#include<list>

class Quadrante{
    private:
        int x, y;
    public:
        Quadrante();
        Quadrante(int, int);
        void setX(int);
        void setY(int);
        int getX();
        int getY();
        void printQuadrante();
};

Quadrante::Quadrante(){}
Quadrante::Quadrante(int x, int y): x{x}, y{y}{}

void Quadrante::setX(int x){
    this->x = x;
}

void Quadrante::setY(int y){
    this->y = y;
}

int Quadrante::getX(){
    return this->x;
}

int Quadrante::getY(){
    return this->y;
}

void Quadrante::printQuadrante(){
    if(this->x>0 && this->y>0)
        std::cout << "primeiro" << std::endl;
    else if(this->x > 0 && this->y<0)
        std::cout << "quarto" << std::endl;
    else if(this->x < 0 && this->y>0)
        std::cout << "segundo" << std::endl;
    else if(this->x < 0 && this->y<0)
        std::cout << "terceiro" << std::endl;
}

int main(){
    int a, b;
    std::list<Quadrante> data;
    do{
        std::cin >> a >> b;
        data.push_back(Quadrante(a, b));
    } while(a!=0 && b!=0);
    
    std::list<Quadrante>::iterator it = data.begin();
    for(it; it!=data.end(); it++){
        it->printQuadrante();
    }
    
    return 0;
}
